server/graphql/v2/query/collection/ExpensesCollectionQuery.ts

```diff
@@ -378,6 +378,47 @@ export const ExpensesCollectionQueryResolver = async (
   return {
     nodes: result.rows,
     totalCount: result.count,
+    totalAmount: () => {
+      const getCurrencyTotalAmountsMemo = (() => {
+        let currencyTotalsPromise = null;
+        return () => {
+          if (!currencyTotalsPromise) {
+            currencyTotalsPromise = (async () => {
+              const query = (await models.Expense.findAll({
+                attributes: [
+                  [Sequelize.col('"Expense"."currency"'), 'expenseCurrency'],
+                  [Sequelize.fn('SUM', Sequelize.col('amount')), 'amount'],
+                ],
+                group: 'expenseCurrency',
+                include,
+                where,
+                raw: true,
+              })) as unknown as { expenseCurrency: string; amount: number }[];
+
+              return query.map(result => ({ currency: result.expenseCurrency, value: result.amount }));
+            })();
+          }
+
+          return currencyTotalsPromise;
+        };
+      })();
+
+      return {
+        amountsByCurrency: getCurrencyTotalAmountsMemo,
+        amount: async ({ currency = 'USD' }) => {
+          const currencyAmounts = await getCurrencyTotalAmountsMemo();
+
+          const values = await req.loaders.CurrencyExchangeRate.convert.loadMany(
+            currencyAmounts.map(v => ({ amount: v.value, fromCurrency: v.currency, toCurrency: currency })),
+          );
```

Comment:
This seems like an unnecessary complication. We could have used `memoize` from lodash to simplify it, but since the promise needs to be fetched in all subfields why not simply can the promise directly?

```suggestion
      const query = (await models.Expense.findAll({
        attributes: [
          [Sequelize.col('"Expense"."currency"'), 'expenseCurrency'],
          [Sequelize.fn('SUM', Sequelize.col('amount')), 'amount'],
        ],
        group: 'expenseCurrency',
        include,
        where,
        raw: true,
      })) as unknown as { expenseCurrency: string; amount: number }[];

      const amountsByCurrency = query.map(result => ({ currency: result.expenseCurrency, value: result.amount }));

      return {
        amountsByCurrency,
        amount: async ({ currency = 'USD' }) => {
          const values = await req.loaders.CurrencyExchangeRate.convert.loadMany(
            amountsByCurrency.map(v => ({ amount: v.value, fromCurrency: v.currency, toCurrency: currency })),
          );
```

Reply:
Yep, you absolutely correct!